### PR TITLE
Implement  Notify REST API > Credentials

### PR DESCRIPTION
**Pull request by mxpv**
Hey. This PR implements API for Credentials resource (https://www.twilio.com/docs/api/notify/rest/credentials)
@kevinburke please take a look on this

### CODE REVIEW

**Comment by kevinburke** [comment]
thanks for this! it might take me a while to get to reviewing and merging,
unfortunately


--
Kevin Burke
925.271.7005 | kev.inburke.com

File: http.go [hunk 1]
```diff
@@ -216,6 +223,27 @@ func NewPricingClient(accountSid string, authToken string, httpClient *http.Clie
 	return c
 }
 
+func NewNotifyClient(accountSid string, authToken string, httpClient *http.Client) *Client {
+	if httpClient == nil {
```
Comment (kevinburke) [inline http.go]: Any reason we can't use the `newNewClient` helper here? I understand the naming is poor...

File: notify_credentials.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"context"
+	"net/url"
+)
+
+type CredentialType string
+
+const credentialsPathPart = "Credentials"
+
+const (
+	TypeGCM = CredentialType("gcm")
```
Comment (kevinburke) [inline notify_credentials.go]: Can you document these or link to the Twilio page where these are documented?

File: notify_credentials.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+type NotifyCredentialPage struct {
+	Page
+	Credentials []*NotifyCredential `json:"credentials"`
+}
+
+type NotifyCredentialPageIterator struct {
+	p *PageIterator
+}
+
+func (n *NotifyCredentialsService) Create(ctx context.Context, data url.Values) (*NotifyCredential, error) {
```
Comment (kevinburke) [inline notify_credentials.go:43]: Would it make sense to add a helper here, where we initialize some defaults, a la `Messages.SendMessage`?

**Comment by kevinburke** [comment]
also you can fix the tests with the following patch in BUILD.bazel:

```patch
diff --git a/BUILD.bazel b/BUILD.bazel
index 5713025..688c0d7 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -23,6 +23,8 @@ go_library(
         "media_go17.go",
         "media_goprevious.go",
         "messages.go",
+        "notify.go",
+        "notify_credentials.go",
         "outgoingcallerids.go",
         "page.go",
         "participants.go",
@@ -62,6 +64,7 @@ go_test(
         "http_test.go",
         "media_test.go",
         "messages_test.go",
+        "notify_credentials_test.go",
         "outgoingcallerids_test.go",
```

File: notify_credentials.go (hunk 3 above)
Comment (mxpv) [reply notify_credentials.go:43]: Done below

**Comment by mxpv** [comment]
Done

File: notify_credentials.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	data.Set("FriendlyName", friendlyName)
+	data.Set("Type", "apn")
+	data.Set("Certificate", cert)
+	data.Set("PrivateKey", privateKey)
+
+	if sandbox {
+		data.Set("Sandbox", "true")
+	} else {
+		data.Set("Sandbox", "false")
+	}
```
Comment (kevinburke) [inline notify_credentials.go:80]: These are great. Thanks.

**Comment by kevinburke** [comment]
Merged as a9136d1adbbf75b5c5b73eacf51fd2e6ea77877a, thanks!

**Pull request closed without merging**